examples/systemcalls/systemcalls.c

```diff
@@ -93,6 +131,53 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
  *
 */
 
+    // Fork the process to create a new child process
+    pid_t pid = fork();
```

Comment:
@Kanin-McGuire how could you reuse logic with the function above and avoid duplication?